### ISSUE TITLE
Design and implement components to edit material values (vectors etc)

### ISSUE THREAD

**Issue by tonihele** [jMonkeyEngine/sdk#600]
Design and implement components to edit material values (vectors etc). I think we are mainly missing editors for matrices and vectors. And int & float arrays and array vectors. It is difficult for a new comer to understand how to input these values. Like the decimal separator is always "." and what is a vector separator etc...

Challenge is to design components that provide ease to use but also efficient for those who'd just prefer to write the value quickly.

These are distributed from com.jme3.gde.materials.multiview.widgets.WidgetFactory.

**Comment by neph1** [comment]
Would make it easier if we could get some reference design. How do others do it?

**Comment by tonihele** [comment]
Yes, that is a very good question. Maybe vectors could use similar to the
colors. Separate text fields for the separate vector components. Also the
very least we could add validation to the components. Like floats are
supposed to be given with "." separator and all that. Prevent wrong inputs
and guide for the right format. Also for vector if we end up having it in
single input.

I did look about masks, but I dont think those are usable in our cases.
Shame.

**Comment by neph1** [comment]
Here's Unity for some inspiration.
![Screenshot from 2024-05-01 07-25-42](https://github.com/jMonkeyEngine/sdk/assets/7988802/a8b790c3-aa70-492f-9634-4134ea519c27)

Having sliders for floats is not a bad idea. There's plenty of space.
Maybe make inserting values into Vector3f manually an extra step. They take up the most space.

**Comment by neph1** [comment]
I've begun with a slider for floats that are bound 0-1.

**Comment by neph1** [comment]
Should we just remove the option to set values for the color? The selector allows fine grained editing of RGB values.

![image](https://github.com/jMonkeyEngine/sdk/assets/7988802/1aca7ccf-2be2-4829-8945-aadbd7dad783)
